### ISSUE TITLE
Error when creating datetime.time with dateutil.tz.tzlocal 

### ISSUE THREAD

**Issue by pganssle** [dateutil/dateutil#59]
Migrating launchpad bug [#1256908](https://bugs.launchpad.net/dateutil/+bug/1256908).

Quoting [Jan Spurny](https://launchpad.net/~jspurny) on 2013-12-02:
> Creating `datetime.time` with `dateutil.tz.tzlocal` creates valid `datetime.time` object, but this object raises error when `__str__`, `__repr__` or `isoformat` method is called on it.

> Demonstration:

     >>> import datetime
     >>> import dateutil.tz
     >>> t1 = datetime.time(1, 2, tzinfo=dateutil.tz.tzutc())
     >>> t2 = datetime.time(1, 2, tzinfo=dateutil.tz.tzlocal())
     >>> t1
     datetime.time(1, 2, tzinfo=tzutc())
     >>> t2
     datetime.time(1, 2, tzinfo=tzlocal())
     >>> print t1
     01:02:00+00:00
     >>> print t2
     Traceback (most recent call last):
       File "<stdin>", line 1, in <module>
       File "/usr/lib/pymodules/python2.6/dateutil/tz.py", line 92, in utcoffset
         if self._isdst(dt):
       File "/usr/lib/pymodules/python2.6/dateutil/tz.py", line 134, in _isdst
         + dt.second)
     AttributeError: 'NoneType' object has no attribute 'toordinal'

> My locale timezone is "Europe/Prague", python2.6, dateutil 1.4.1, debian linux.

**Comment by pganssle** [comment]
This bug is confirmed on Windows, Python 2.7.9, Central Time Zone.

**Comment by pganssle** [comment]
I believe this is a manifestation of #132, which was fixed by #181.

I can no longer reproduce this in Linux.

**Comment by labrys** [comment]
@pganssle Checked with Python 2.7.11 and 3.5.1 on Win 10.
While this issue is fixed, while testing I found a related issue:
```python
>>> t3 = datetime.time(1, 2, tzinfo=tz.tzwinlocal())
Traceback (most recent call last):
  File "<pyshell#8>", line 1, in <module>
    print(t3)
  File "C:\Python35\lib\site-packages\dateutil\tz\win.py", line 135, in utcoffset
    if self._isdst(dt):
  File "C:\Python35\lib\site-packages\dateutil\tz\win.py", line 172, in _isdst
    dston = picknthweekday(dt.year, self._dstmonth, self._dstdayofweek,
AttributeError: 'NoneType' object has no attribute 'year'
```
I'll create a new issue for it.
